packages/@lwc/template-compiler/src/__tests__/fixtures/directive-for-each/foreach-with-if/expected.js

```diff
@@ -0,0 +1,31 @@
+import { registerTemplate } from "lwc";
+function tmpl($api, $cmp, $slotset, $ctx) {
+  const {
+    k: api_key,
+    t: api_text,
+    d: api_dynamic,
+    h: api_element,
+    i: api_iterator,
+  } = $api;
+  return [
+    api_element(
+      "section",
+      {
+        key: 0,
+      },
+      api_iterator($cmp.items, function (item) {
```

Comment:
The behavior to note is that iterator directive is processed first and then the if directive.